src/agent/communicator/CMakeLists.txt

```diff
@@ -14,9 +14,12 @@ find_package(nlohmann_json CONFIG REQUIRED)
 find_path(JWT_CPP_INCLUDE_DIRS "jwt-cpp/base.h")
 
 add_library(Communicator src/communicator.cpp src/http_client.cpp)
-target_include_directories(Communicator PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include ${JWT_CPP_INCLUDE_DIRS})
+target_include_directories(Communicator PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include SYSTEM INTERFACE ${JWT_CPP_INCLUDE_DIRS})
```

Comment:
What bout the `src` folder?

Reply:
It's not necessary to include the folder if using `""` instead of `<>` for including files in the same directory.